sklearn/impute/tests/test_knn.py

```diff
@@ -545,3 +545,42 @@ def test_knn_imputer_distance_weighted_not_enough_neighbors(na, working_memory):
 def test_knn_tags(na, allow_nan):
     knn = KNNImputer(missing_values=na)
     assert knn._get_tags()["allow_nan"] == allow_nan
+
+
+@pytest.mark.parametrize("add_indicator", [True, False])
+@pytest.mark.parametrize("missing_value_test", [np.NaN, 1])
+def test_knn_imputation_adds_missing_indicator_column_if_add_indicator_is_true(
```

Comment:
I think this property applies to all imputers, thus I think it's better to have a test in `sklearn/impute/tests/test_common.py`:

```python
@pytest.mark.parametrize("imputer", imputers(), ids=lambda x: x.__class__.__name__)
@pytest.mark.parametrize("missing_value_test", [np.nan, 1])
def test_imputation_adds_missing_indicator_if_add_indicator_is_true(
    imputer, missing_value_test
):
    """Check that missing indicator always exists when add_indicator=True.

    Non-regression test for gh-26590.
    """
    X_train = np.array([[0, np.NaN], [1, 2]])

    # Test data without missing values that can be set to np.NaN or 1.
    X_test = np.array([[0, missing_value_test], [1, 2]])

    imputer.set_params(add_indicator=True)
    imputer.fit(X_train)

    X_test_imputed_with_indicator = imputer.transform(X_test)
    assert X_test_imputed_with_indicator.shape == (2, 3)

    imputer.set_params(add_indicator=False)
    imputer.fit(X_train)
    X_test_imputed_without_indicator = imputer.transform(X_test)
    assert X_test_imputed_without_indicator.shape == (2, 2)

    assert_allclose(
        X_test_imputed_with_indicator[:, :-1], X_test_imputed_without_indicator
    )
    if np.isnan(missing_value_test):
        expected_missing_indicator = [1, 0]
    else:
        expected_missing_indicator = [0, 0]

    assert_allclose(X_test_imputed_with_indicator[:, -1], expected_missing_indicator)

```

Given the original bug report, I am okay with checking `add_indicator=True`, which is also consistent with the name of the test "*_if_add_indicator_is_true".